devops/jobs/ImageBuilder.groovy

```diff
@@ -26,54 +26,76 @@ import static org.edx.jenkins.dsl.Constants.common_wrappers
 
 class ImageBuilder {
     public static def job = { dslFactory, extraVars ->
-        dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/image-builder") {
+        //assertions come earlier to ensure APPS_TO_CONFIG can be read
+        assert extraVars.containsKey('APPS_TO_CONFIG'): "Please define APPS_TO_CONFIG"
+        assert extraVars.containsKey('CONFIGURATION_REPO_URL') : "Please define CONFIGURATION_REPO_URL"
 
-            assert extraVars.containsKey('CONFIGURATION_REPO_URL') : "Please define CONFIGURATION_REPO_URL"
+        // for each application
+        extraVars.get("APPS_TO_CONFIG").each { app_name, app_config ->
+            dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/image-builders/" + app_name + "-image-builder") {
 
-            def config_branch = 'master'
+                def config_branch = 'master'
 
-            wrappers common_wrappers
-            // add credentials to log in to DockerHub; ID refers to credential ID as set up in Jenkins
-            wrappers {
-                credentialsBinding {
-                    file("CONFIG_JSON_FILE", extraVars.get("CONFIG_JSON_FILE_CREDENTIAL_ID", "docker-config-json"))
+                // inject APP_NAME as an environment variable for use by build-push-app.sh
+                environmentVariables {
+                    env("APP_NAME", app_name)
                 }
-            }
-            logRotator common_logrotator
 
-            // add APP_NAME as parameter; will be passed in from upstream job as application Dockerfile to build
-            parameters{
-                stringParam('APP_NAME')
-            }
+                wrappers common_wrappers
+                // add credentials to log in to DockerHub; ID refers to credential ID as set up in Jenkins
+                wrappers {
+                    credentialsBinding {
+                        file("CONFIG_JSON_FILE", extraVars.get("CONFIG_JSON_FILE_CREDENTIAL_ID", "docker-config-json"))
+                    }
+                    // if the application has a repository, the build name will display the version of the application used; otherwise, display
+                    // only the configuration version
+                    if (app_config.get("app_repo")) {
```

Comment:
It doesn't make sense to have "app. version = " for applications without application code repositories, so we have different buildName structures depending upon whether the application has a code repository or not.